src/NormalDepthMap.cpp

```diff
@@ -73,10 +53,39 @@ bool NormalDepthMap::isDrawNormal() {
 
 void NormalDepthMap::setDrawDepth(bool drawDepth) {
     _drawDepth = drawDepth;
+    if (_normalDepthMapNode != NULL)
+        _normalDepthMapNode->getOrCreateStateSet()->getUniform("drawDepth")->set(_drawDepth);
+}
+
+void NormalDepthMap::addNodeChild(osg::ref_ptr<osg::Node> node) {
+    _normalDepthMapNode->addChild(node);
 }
 
 bool NormalDepthMap::isDrawDepth() {
     return _drawDepth;
 }
 
+void NormalDepthMap::createTheNormalDepthMapShaderNode() {
```

Comment:
A tiny comment, but if I were you I would create a side-effect-free createShaderNode (which would *return* the osg::Group) and then attach that node to the tree elsewhere. Concentrating the side effects is always a good idea for comprehension